### PR TITLE
cb-61

### PR DESCRIPTION
**Pull request by axrs**

### CODE REVIEW

File: src/cljc/centrebull/spec.cljc [hunk 1]
```diff
@@ -72,6 +73,48 @@
...
 
+(defn- shot-length-10-15
+  "Ensures the shots are either 10 or 15 shot matches (with 2 optional siders)"
+  [s]
+  (some #{(count s)} [10 11 12 15 16 17]))
+
+(defn- valid-shot-chars-only
+  "Ensures the shots are either -0123456V or X"
+  [s]
+  (if (empty? (filter #(not (some #{%} "-0123456VX")) s))
```
Comment (CalebMacdonaldBlack) [inline src/cljc/centrebull/spec.cljc]: just return empty

File: src/cljc/centrebull/spec.cljc [hunk 2]
```diff
@@ -72,6 +73,48 @@
...
+        len (if (< 12 c) 15 10)]
+    (->> (min c len)
+      (subs s 0)
+      (map shot->int)
+      (apply +))))
+
+(defn- calculate-vs [s]
+  (count (filter #(= \V %) s)))
+
+(defn- calcualte-result [{:keys [result/shots]
```
Comment (CalebMacdonaldBlack) [inline src/cljc/centrebull/spec.cljc]: spelling error 📖

File: src/cljc/centrebull/spec.cljc [hunk 3]
```diff
@@ -72,6 +73,48 @@
...
+                          :or   {shots ""}
+                          :as   m}]
+  (-> m
```
Comment (CalebMacdonaldBlack) [inline src/cljc/centrebull/spec.cljc]: why bother?
`(merge m {:result/score (calculate-score shots) :result/vs (calculate-vs shots)})`

File: src/cljc/centrebull/spec.cljc [hunk 4]
```diff
@@ -109,6 +152,12 @@
 
 (s/def :search/q string?)
 
+(s/def :result/id (s/conformer ->uuid))
+(s/def :result/shots (s/and string?
+                       shot-length-10-15
+                       valid-shot-chars-only))
+;(s/conformer calcualte-result)))
```
Comment (CalebMacdonaldBlack) [inline src/cljc/centrebull/spec.cljc]: remove comment

**Review by CalebMacdonaldBlack — CHANGES_REQUESTED** [review]

**Review by deanditton — APPROVED** [review]

**Review by CalebMacdonaldBlack — APPROVED** [review]

**Pull request merged**